### ISSUE TITLE
tensorflow_addons.text.crf_log_likelihood could not work on gpu thread

### ISSUE THREAD

**Issue by xexiyong** [tensorflow/addons#2454]
**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):     Linux centOS
- TensorFlow version and how it was installed (source or binary):     2.4.1  pip install binary
- TensorFlow-Addons version and how it was installed (source or binary):      0.12.0  pip install binary
- Python version:    3.7.3
- Is GPU used? (yes/no):  yes

**Describe the bug**

to upgrade the tf 1.0 old NER task model,  from the tf timeline's training op duration, the crf_log_likelihood function seems only work on the cpu thread, and when its running on the cpu thread, the other gpu operation have to wait. so the training process consumes several seconds while only less than 1 seconds on the tf 1.0 old model. 

![image](https://user-images.githubusercontent.com/12406838/115549323-18a5f900-a2db-11eb-8b27-02fa780df2c8.png)

so How could I move this function into the gpu thread?

**Code to reproduce the issue**

the code is a little complicated, while I think the issue could be open for a long period. Thank you.

**Other info / logs**

I have the trace info, and I think I could be active for a long time in this issue.

**Comment by xexiyong** [comment]
and the tfa-nightly does not solve this problem

**Comment by xexiyong** [comment]
I found the tf.cond op which in crf_log_likelihood function is working on cpu thread, it causes this, delete it, training speed is better!

**Comment by xexiyong** [comment]
@bhack is this a bug in tf? the tf.cond in tf 1.4 is working in gpu thread, is it possible to recover this property?

**Comment by bhack** [comment]
Can you point to the specific line in the repository?

**Comment by xexiyong** [comment]
https://github.com/tensorflow/addons/blob/53ef58f3aac9f3e1c36fca3bccaec73f67cd491c/tensorflow_addons/text/crf.py#L104
https://github.com/tensorflow/addons/blob/53ef58f3aac9f3e1c36fca3bccaec73f67cd491c/tensorflow_addons/text/crf.py#L201

**Comment by bhack** [comment]
Can you log the device placement? https://www.tensorflow.org/guide/gpu#logging_device_placement

**Comment by xexiyong** [comment]
> Can you log the device placement? https://www.tensorflow.org/guide/gpu#logging_device_placement

this seems useless..

**Comment by howl-anderson** [comment]
@bhack Please also label this issue with `crf`. So, we can find it by `crf` in label filter.

**Comment by seanpmorgan** [comment]
TensorFlow Addons is transitioning to a minimal maintenance and release mode. New features will not be added to this repository. For more information, please see our public messaging on this decision:
[TensorFlow Addons Wind Down](https://github.com/tensorflow/addons/issues/2807)

Please consider sending feature requests / contributions to other repositories in the TF community with a similar charters to TFA:
[Keras](https://github.com/keras-team/keras)
[Keras-CV](https://github.com/keras-team/keras-cv)
[Keras-NLP](https://github.com/keras-team/keras-nlp)
